File: Versión Español/Estructuras de Datos/Arrays/Arrays.php
<?php
/**
 * @author Jose Chavarría
 * @license MIT
 * @link https://chavarria.cr/post/arrays
 */

class Arrays 
{
	function __construct()
	{
		$arrayGeneral = array();
		$this->Insertar();
	}

	function Insertar(){
		echo("Insertar...");
		for ($i=0; $i <5; $i++) { 
			echo '</br>'.'índice: '.$i.' valor: '.$this->arrayGeneral[]=$i;
		}
	}

	function Recorrido(){
		echo("</br></br>Recorrido...");
		for ($i=0; $i < count($this->arrayGeneral); $i++) { 
			echo '</br>'.'índice: '.$i.' valor: '.$this->arrayGeneral[$i];
		}
	}

	function Actializar($past, $var){
		echo ("</br></br>Actializar: ".$past." a ".$var);
		for ($i=0; $i < count($this->arrayGeneral); $i++) { 
			if($this->arrayGeneral[$i]==$past){
				echo '</br>'.'índice: '.$i.' valor: '.$this->arrayGeneral[$i] =$var;
			}
		}
	}
	function Buscar($var){
		echo ("</br></br>Buscar :".$var);
		for ($i=0; $i < count($this->arrayGeneral); $i++) { 
			if ($this->arrayGeneral[$i]==$var) {
				echo '</br>'.'índice: '.$i.' valor: '.$this->arrayGeneral[$i];
			} 
		}
	}
	function Eleminacion($var){
		echo ("</br></br>Eleminacion: ".$var."</br>");
		$nuevoArray = array(); $aux=0;
		for ($i=0; $i < count($this->arrayGeneral) ;$i++) { 
			if($this->arrayGeneral[$i]!=$var){
				$this->nuevoArray[$aux]=$this->arrayGeneral[$i];
				$aux++;
			}
		}
		$this->arrayGeneral = $this->nuevoArray;
	}
}

$a = new Arrays();
$a->Recorrido();
$a->Buscar(2);
$a->Actializar(2,100);
$a->Eleminacion(3);
$a->Recorrido();

?>

File: Versión Español/Estructuras de Datos/Arrays/Arrays.java
package arrays;

/**
 * @author Jose Chavarría
 * @license MIT
 * @link https://chavarria.cr/post/arrays
 */
public class Arrays {

    /**
     * @param args the command line arguments
     */
    static int[] num = new int[5];

    static void Insercion() {
        System.out.println("Inserción...");
        for (int i = 0; i < num.length; i++) {
            num[i] = i + 10;
            System.out.println(num[i] = i + 10);
        }
    }

    static void Recorrido() {
        System.out.println("\nRecorrido...");
        for (int i = 0; i < num.length; i++) {
            System.out.println("índice: " + i + " elemento: " + num[i]);
        }
    }

    static void Buscar(int var) {
        System.out.println("\nBúsqueda: " + var);
        for (int i = 0; i < num.length; i++) {
            if (num[i] == var) {
                System.out.println("índice: " + i + " elemento: " + num[i]);
            }
        }
    }

    static void Actualizar(int pasado, int nuevo) {
        System.out.println("\nActualizar: " + pasado + " por " + nuevo);
        for (int i = 0; i < num.length; i++) {
            if (num[i] == pasado) {
                num[i] = nuevo;
                System.out.println("índice: " + i + " elemento: " + num[i]);
            }
        }
    }

    static void Eliminar(int var) {
        System.out.println("\nEliminación: " + var);
        int[] temp = new int[4];
        int contador = 0, aux = 0;

        while (contador < num.length) {
            if (num[contador] != var) {
                temp[aux] = num[contador];
                aux++;
            }
            contador++;
        }
        num = temp;
    }

    public static void main(String[] args) {
        Insercion();
        Recorrido();
        Buscar(10);
        Actualizar(10, 100);
        Eliminar(100);
        Recorrido();
    }
}
